climada/hazard/centroids/centr.py

```diff
@@ -555,60 +555,47 @@ def union(self, *others):
         ------
             ValueError
         """
-        cent_list = list(others)
-        if self.lat.any() or self.meta:
-            cent_list.insert(0, self)
-
-        if np.all([cent_list[0].equal(cent) for cent in cent_list[1:]]):
-            return copy.deepcopy(cent_list[-1])
-
-        #Set attributes to centroids if missing in one but defined in other
-        is_area_pixel = np.any([cent.area_pixel.any() for cent in cent_list]) \
-            and not np.all([cent.area_pixel.any() for cent in cent_list])
-        is_dist_coast = np.any([cent.dist_coast.any() for cent in cent_list]) \
-            and not np.all([cent.dist_coast.any() for cent in cent_list])
-        is_on_land = np.any([len(cent.on_land) > 0 for cent in cent_list]) \
-            and not np.all([len(cent.on_land) > 0 for cent in cent_list])
-        is_region_id = np.any([cent.region_id.any() for cent in cent_list]) \
-            and not np.all([cent.region_id.any() for cent in cent_list])
-        is_elevation = np.any([cent.elevation.any() for cent in cent_list]) \
-            and not np.all([cent.elevation.any() for cent in cent_list])
+        # restrict to non-empty centroids
+        cent_list = [c for c in (self,) + others if c.size > 0 or c.meta]
+        if len(cent_list) == 0 or len(others) == 0:
+            return copy.deepcopy(self)
 
+        # check if all centroids agree
+        if all([cent_list[0].equal(cent) for cent in cent_list[1:]]):
```

Comment:
I think this will fail if a centroids as no defined `crs` (which was one of the reasons to `set_geometry_points`). This was addressed by initializing centroids with a default `crs`. However, older data might still fail. Probably not a big deal.

Reply:
This shouldn't fail because there is now a fallback CRS in `Centroids.crs`: https://github.com/CLIMADA-project/climada_python/pull/268/files#diff-0ec37b2aaa4e9ad606f7859d74412d0af63acc0712fbc49baed2a7935487e4edR1013-R1015